pandas/tests/io/formats/test_format.py

```diff
@@ -3240,14 +3240,29 @@ def test_repr_html_ipython_config(ip):
 
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
+@pytest.mark.parametrize("encoding", [None, "utf-8", "gbk", "foo"])
 def test_filepath_or_buffer_arg(
-    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals
+    float_frame,
+    method,
+    filepath_or_buffer,
+    assert_filepath_or_buffer_equals,
+    encoding,
+    filepath_or_buffer_id,
 ):
     df = float_frame
-    expected = getattr(df, method)()
 
-    getattr(df, method)(buf=filepath_or_buffer)
-    assert_filepath_or_buffer_equals(expected)
+    if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
+        with pytest.raises(
+            ValueError, match="buf is not a file name and encoding is specified."
+        ):
+            getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)
+    elif encoding == "foo":
+        with pytest.raises(LookupError, match="unknown encoding"):
+            getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)
+    else:
+        expected = getattr(df, method)()
+        getattr(df, method)(buf=filepath_or_buffer, encoding=encoding)
+        assert_filepath_or_buffer_equals(expected)
```

Comment:
Is this actually asserting anything? It doesn't seem to respect the encoding when reading back in; I think need a test case where the encoding actually makes a difference and validate on read/write

Reply:
When the test is split into two tests `test_filepath_or_buffer_arg` and `test_filepath_or_buffer_arg_raises` , I think this won't be a problem